In [35]:
from pymatgen.ext.matproj import MPRester
from pymatgen.analysis.diffraction.xrd import XRDCalculator
from IPython.display import display
from pymatgen.electronic_structure.plotter import BSPlotter
import matplotlib.pyplot as plt


with MPRester("o1xT73gLJ9EAfjhPW2") as m:
    structure = m.get_bandstructure_by_material_id("mp-14116")

E=structure.efermi
band_gap=structure.get_band_gap()
direct_band_gap=structure.get_direct_band_gap()
CBM=structure.get_cbm()
VBM=structure.get_vbm()

## Bande interdite

In [48]:
print(band_gap)

{'direct': False, 'energy': 0.9581, 'transition': '(0.634,0.366,0.000)-(0.608,0.608,0.284)'}


En analysant les informations sur la bande interdite ci-dessus, on en conclu que la bande interdite est indirecte (on peut clairement le voir à la figure du point suivant).
On remarque égalemment que l'énergie de la bande interdite est de 0.9581 eV, le CuRhO2 est donc un semi-conducteur.

## Directions de dispertion

Les endroits de la dernière (première) bande de valence (conduction) où la dispersion est la plus forte et la moins forte sont les endroits où les pentes sont les plus et moins fortes, ces directions sont représentées par des flèches (vertes pour les moins fortes dispersions, orange pour les plus fortes) sur le graphe ci dessous.

![im2](symmetry/pentes2.png)

## Masse effective

En théorie, comme on suppose une dispersion parabloique, la masse effective est constante et peut être calculée avec la formule suivante : $$ m^* = \frac{ \bar{h}^2}{ d^2 \epsilon / dk^2} $$.

En pratique, pour des états voisins d'un minimum ou d'un maximum de bande, la relation $\epsilon(k)$ peut souvent être approximée comme suit : 

$$ \epsilon(k)-\epsilon_0  =  \frac{ \bar{h}^2 }{ 2m^*} (k_x^2 + k_y^2 + k_z^2)$$

In [60]:
h= (6.58e-16)**2 #eV/s

masse_conduc=((h)*(CBM.get('kpoint').cart_coords[0]**2 + CBM.get('kpoint').cart_coords[1]**2 + CBM.get('kpoint').cart_coords[2]**2))/2*(CBM.get('energy')-E)

masse_valence=((h)*(VBM.get('kpoint').cart_coords[0]**2 + VBM.get('kpoint').cart_coords[1]**2 + VBM.get('kpoint').cart_coords[2]**2))/2*(VBM.get('energy')+E)

print(masse_conduc)
print(masse_valence)

1.7627703728145316e-31
3.165295103258282e-30


A l'aide de cette deuxième équation, j'ai pu trouver la masse effective car nous avons les énergies aux sommet (base) de la dernière (première) bande de valence (conduction). On obtient des valeurs de :  
1.7627703728145316e-31 kg pour le sommet de la bande de conduction  
3.165295103258282e-30 kg pour la base de la bande de valence